### PR TITLE
chore(ci/debugger): increase range for flaky test

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
This test run:

```
            (snapshot,) = d.test_queue
>           assert 0.9 * duration <= snapshot.duration <= 2.5 * duration, snapshot
E           AssertionError: Snapshot(probe=FunctionProbe(probe_id='duration-probe', tags={}, active=True, rate=1.0, condition=None, module='tests.... 'throwable': None}, duration=284068, timestamp=1668637423.0604956, snapshot_id='0b0bba8c-dfaf-4884-9233-3b3208bfbaa1')
E           assert 284068 <= (2.5 * 100000.0)
E            +  where 284068 = Snapshot(probe=FunctionProbe(probe_id='duration-probe', tags={}, active=True, rate=1.0, condition=None, module='tests.... 'throwable': None}, duration=284068, timestamp=1668637423.0604956, snapshot_id='0b0bba8c-dfaf-4884-9233-3b3208bfbaa1').duration
```
https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/24048/workflows/07d39bba-30fd-49d5-a5e8-ec85f58ad8c5/jobs/1633200

demonstrates that a factor of 2.5 is too narrow. Increase it to 10.0.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Comment by Kyle-Verhoog** [comment]
https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/24068/workflows/320b950f-d00f-4461-9195-872327dff816/jobs/1634424 Might have to be even higher than `3.0`

**Comment by Kyle-Verhoog** [comment]
https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/24082/workflows/c3940aaa-d3e3-480a-98db-044911f137f0/jobs/1635454 maybe even higher than `4.0` 😱

**Comment by P403n1x87** [comment]
@Kyle-Verhoog thanks for taking care of this! This is tricky, as I think the OS gives you guarantees only on the fact that it will sleep _at least_ for the requested time interval. I wouldn't mind setting the factor to 10, or removing the upper-bound check altogether BTH, given the nuisance this is causing.

File: tests/debugging/test_debugger.py [hunk 1]
```diff
@@ -698,7 +698,7 @@ def test_debugger_function_probe_duration(duration):
         durationstuff(duration)
 
         (snapshot,) = d.test_queue
-        assert 0.9 * duration <= snapshot.duration <= 2.5 * duration, snapshot
+        assert 0.9 * duration <= snapshot.duration <= 3.0 * duration, snapshot
```
Comment (P403n1x87) [inline tests/debugging/test_debugger.py:701]:
```suggestion
        assert 0.9 * duration <= snapshot.duration <= 10.0 * duration, snapshot
```

**Review by brettlangdon — APPROVED** [review]
I took a different approach, which was to use a pytest plugin to re-run on failure.

**Pull request merged**